modules/services/getmail.nix

```diff
@@ -0,0 +1,205 @@
+        description = ''
+          The destination path. For <literal>Maildir</literal> it's the file
+          path and for <literal>MDA_external</literal> it's the destination
+          application.
+        '';
+      };
+    };
+  });
+
+  optionsModule = types.submodule ({config,...}: {
+    options = {
+      delete = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Enable if you want to delete read messages from the server. Most
+          users should either enable <literal>delete</literal> or disable
+          <literal>readAll</literal>.
+        '';
+      };
+
+      readAll = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Enable if you want to fetch all, even the read messages from the
+          server. Most users should either enable <literal>delete</literal> or
+          disable <literal>readAll</literal>.
+        '';
+      };
+    };
+  });
+
+in
+
+{
+  options = {
+    programs.getmail = {
+      enable = mkEnableOption "Enable getmail";
+
+      retriever = mkOption {
+        type = retrieverModule;
+        default = {};
+        description = "The server section.";
+      };
+
+      destination = mkOption {
+        type = destinationModule;
+        default = {};
+        description = "The destination section.";
+      };
+
+      options = mkOption {
+        type = optionsModule;
+        default = {};
+        description = "The options section.";
+      };
+
+      frequency = mkOption {
+        type = types.string;
+        default = "*:0/15";
+        example = "hourly";
+        description = ''
+          The refresh frequency. Check <literal>man systemd.time</literal> for
+          more information on the syntax.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.file.".getmail/getmailrc".text =
+    let
+      quoted = x: "\"${escape ["\""] x}\"";
+
+      passwordCommand = concatStringsSep ", " (map quoted cfg.retriever.passwordCommand);
+
+      password = if cfg.retriever.passwordCommand != null
+        then "password_command = (${passwordCommand})"
+        else optionalString (cfg.retriever.password != null) "password = \"${quoted cfg.retriever.password}\"";
+      mailboxInner = concatStringsSep ", " (
+        map quoted cfg.retriever.mailboxes);
+
+      mailboxes = "(${mailboxInner})";
+
+      in
+
+    ''
+      [retriever]
+      type = ${cfg.retriever.type}
+      server = ${cfg.retriever.server}
+      username = ${cfg.retriever.username}
+      ${password}
+      mailboxes = ${mailboxes}
+
+      [destination]
+      type = ${cfg.destination.type}
+      path = ${cfg.destination.path}
+
+      [options]
+      delete = ${toString cfg.options.delete}
+      read_all = ${toString cfg.options.readAll}
+    '';
+
+    systemd.user.services.getmail = {
+      Unit = {
+        Description = "getmail email fetcher";
+        PartOf = ["network-online.target"];
```

Comment:
This doesn't do anything, because the `network-online.target` isn't present in the users systemd, only the systems one.